lib/DeviceManagerPlugin.ts

```diff
@@ -247,24 +198,12 @@ export class DeviceManagerPlugin extends Plugin {
   registerDecoder (
     decoder: Decoder,
     { handler }: { handler?: PayloadHandler } = {}
-  ): { controller: string, action: string } {
-    decoder.action = decoder.action || Inflector.kebabCase(decoder.deviceModel);
-
-    if (this.api['device-manager/payload'].actions[decoder.action]) {
-      throw new PluginImplementationError(`A decoder for "${decoder.deviceModel}" has already been registered.`);
+  ) {
+    if (! this.decodersService) {
+      this.decoderQueue.push({ decoder, handler });
+    } else {
+      return this.decodersService.register(decoder, { handler });
```

Comment:
This shouldn't be possible and should raise an exception. Even if the decoder is register, the controller action will not be added to Kuzzle